### PR TITLE
Add method to get the featureToggle definition

### PR DESCRIPTION
**Pull request by vsandvold**

### CODE REVIEW

**Comment by vsandvold** [comment]
It works, preferably without `getFeatureToggleDefinition` in the interface, just as a public method on `DefaultUnleash. Otherwise, we must implement it in `FinnUnleash` aswell.

Another caveat with `FinnUnleash` is that we must cast the reference everywhere we want to use the `startExperiment` method. Maybe we should add it to the interface right away, and provide a default implementation for backward compatibility?

**Comment by SimenB** [comment]
>Maybe we should add it to the interface right away, and provide a default implementation for backward compatibility?

👍

File: src/main/java/no/finn/unleash/FakeUnleash.java [hunk 1]
```diff
@@ -24,6 +24,10 @@ public boolean isEnabled(String toggleName, boolean defaultSetting) {
         }
     }
 
+    public FeatureToggle getFeatureToggleDefinition(String toggleName) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/FakeUnleash.java]: can be dropped as it is not part of the interface!

**Comment by ivarconr** [comment]
> Maybe we should add it to the interface right away, and provide a default implementation for backward compatibility?

I think we should do that as soon as we also can ship a working `abTestStrategy` we are happy with!

File: src/main/java/no/finn/unleash/DefaultUnleash.java [hunk 2]
```diff
@@ -92,6 +92,10 @@ public boolean isEnabled(final String toggleName, final UnleashContext context ,
         return enabled;
     }
 
+    public FeatureToggle getFeatureToggleDefinition(String toggleName) {
+        return toggleRepository.getToggle(toggleName);
```
Comment (vsandvold) [inline src/main/java/no/finn/unleash/DefaultUnleash.java]: Is this method nullable, in case of a non-existent toggleName?

Comment (ivarconr) [inline src/main/java/no/finn/unleash/DefaultUnleash.java]: then it will return null, could preferably be wrapped in to an optional perhaps?

**Comment by vsandvold** [comment]
That could work, I'll fix it tomorrow

Den man. 26. jun. 2017, 21.13 skrev Ivar Conradi Østhus <
notifications@github.com>:

**Pull request merged**